config-service-api/src/main/proto/org/hypertrace/config/service/v1/config_service.proto

```diff
@@ -38,6 +38,9 @@ message UpsertConfigRequest {
   // optional - only required if config applies to a specific context.
   // If empty, specified config is associated with a default context.
   string context = 4;
+
+  //optional - only required when we want to have optimistic locking for update
```

Comment:
Can you describe how this behaves if no config exists (the create side of upsert)? Does it always pass? always fail? And the optimistic locking is more of a use case description, i'd drop it.

```suggestion
  // an optional condition that must pass in order for an update operation to succeed. If the target record does not exist...
```

Reply:
i am think to have this conditional update only when the Filter field is present (which uses doc store update method)

other case like create and update with no filter can work as it is (current uses doc store upsert method which is deprecated i can use createOrReplace method instead)

Reply:
makes sense. Just need to reject the create + filter case with validation